### ISSUE TITLE
[Bug]: sharey argument sets up wrong limits with log scale

### ISSUE THREAD

**Issue by Miguel-LlamasLanza** [matplotlib/matplotlib#27156]
### Bug summary

When using sharey in subplots, together with log scale, some part of the plot is not displayed (i.e. the ylims are smaller than it should be) 

### Code for reproduction

```python
import matplotlib.pyplot as plt
import numpy as np

# Create array of values between 0 and 1
arr_1 = np.random.rand(1000)

# Create a second array which have a 10% of values between 1 and 2
arr_2_extra = np.random.rand(100) + 1
arr_2 = np.concatenate([np.random.rand(900), arr_2_extra])

# Let's plot the histogram of both arrays in two adjacent subplots.
fig, axes = plt.subplots(1, 2)
axes[0].hist(arr_1, histtype = 'step')
axes[1].hist(arr_2,  histtype = 'step')

# Let's do the same but with the sharey parameter (Works as expected)
fig, axes = plt.subplots(1, 2, sharey = 'row')
axes[0].hist(arr_1, histtype = 'step')
axes[1].hist(arr_2,  histtype = 'step')

# Now we plot with log scale, but without the sharey parameter (Works good as well)
fig, axes = plt.subplots(1, 2)
[axes[i].set_yscale('log') for i in range(2)]
axes[0].hist(arr_1, histtype = 'step')
axes[1].hist(arr_2,  histtype = 'step')

# Now the plot with sharey parameter and with log-scale (limits are wrong!!)
fig, axes = plt.subplots(1, 2, sharey = 'row')
[axes[i].set_yscale('log') for i in range(2)]
axes[0].hist(arr_1, histtype = 'step')
axes[1].hist(arr_2,  histtype = 'step')
```

### Actual outcome

![image](https://github.com/matplotlib/matplotlib/assets/64196642/59d397cc-b8a5-402e-9e8b-eef2f57b5191)

### Expected outcome

I would expect the limits of the second subplot, in the last case to be equal to the limits of the second subplots in all the other cases. i.e. I would expect the data not to be cut below ~7. It seems a problem with the ylim values

### Additional information

It seems a problem with the ylim values

### Operating system

Ubuntu 22.04

### Matplotlib Version

3.7.1

### Matplotlib Backend

module://matplotlib_inline.backend_inline

### Python version

3.10.11

### Jupyter version

6.5.4

### Installation

conda

**Comment by rcomer** [comment]
I can reproduce this with v3.7.1 but with v3.8.0 I get this for the final pair.  @Miguel-LlamasLanza are you able to try upgrading to the latest release of Matplotlib?
![test](https://github.com/matplotlib/matplotlib/assets/10599679/7ba7b787-e716-47c2-baf5-bff5a4733a95)

**Comment by rcomer** [comment]
Actually this looks like a duplicate of #14527, which was indeed fixed for v3.8.0.  So I'm going to close this.
